python/setup.py

```diff
@@ -30,7 +30,11 @@
 else:
     from setuptools import setup
     from setuptools.extension import Extension
-    kwargs = {'install_requires': ['numpy>=1.17.0,<2.0.0', 'requests>=2.20.0,<3', 'graphviz<0.9.0,>=0.8.1'], 'zip_safe': False}
+    kwargs = {
+        'install_requires': ['requests>=2.20.0,<3', 'graphviz<0.9.0,>=0.8.1']
+            .append('numpy>=1.17.0,<2.0.0' if sys.version_info[0] > 2 else 'numpy>1.16.0,<2.0.0'),
```

Comment:
Why not use https://www.python.org/dev/peps/pep-0508/#environment-markers ?

Reply:
Thanks for pointing this out. I'm not familiar with the tool, but will adopt it in this PR after CI passes.